test/functional/p2p_invalid_getdata.py

```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env python3
+# Copyright (c) 2018 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Request non-existent block.
+"""
+
+from test_framework.mininode import CInv, P2PInterface, msg_getdata
+from test_framework.test_framework import BitcoinTestFramework
+
+class InvalidGetDataTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.setup_clean_chain = True
+        self.num_nodes = 1
+
+    def run_test(self):
+        # Setup the p2p connections and start up the network thread.
+        self.test_node = self.nodes[0].add_p2p_connection(P2PInterface())
+
+        self.test_node.wait_for_verack()
+
+        self.test_node.send_message(msg_getdata(inv=[CInv(2, 0)]))
```

Comment:
What is this testing? I don't see any assertions.

Reply:
I think an earlier version of the code would crash when receiving such an inv. But yeah, somewhat agree that a new test file with all the overhead is a bit too much for testing so little